---
title: "individual_assembly"
author: "sophiaaredas"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  html_document:
    code_folding: show
    highlight: default
    keep_md: yes
    theme: journal
    toc: yes
    toc_float:
      collapsed: no
      smooth_scroll: yes
      toc_depth: 3
  pdf_document:
    toc: yes
editor_options:
  chunk_output_type: console
---
```{=html}
<style>
pre code, pre, code {
  white-space: pre !important;
  overflow-x: scroll !important;
  word-break: keep-all !important;
  word-wrap: initial !important;
}
</style>
```
```{r setup, include=FALSE}
# For width of code chunks and scroll bar
options(width=250)
knitr::opts_chunk$set(eval = TRUE,
                      echo = TRUE,
                      include = TRUE,
                      warning = FALSE,
                      collapse = FALSE,
                      message = FALSE,
                      dpi=200, dev = "png",
                      engine = "R", # Chunks will always have R code, unless noted
                      error = TRUE,
                      fig.path="figures/",
                      fig.align = "center")
```

```{r load-packages}
# Efficiently load packages
pacman::p_load(readr, stringr, tidyr, tidyverse, dplyr, ggplot2, tidytext, scales, wesanderson, ggpubr, wacolors, rcartocolor, treeio, ggtreeExtra, phyloseq, ggtree, ggstar, install = FALSE)
```
## Final Individual Community Analysis

#### So our first analysis with geNomad was with our MAG files which are short reads binned together to make MAGs

#### But we before we created our MAGs we have long contig files before we binned them and put them into MAGs

#### We will see if geNomad does a more accurate prediction of plasmids and phages with the longer contigs pre binned

### So geNomad has finished its analysis and now we have to organize our outputs

#### We will be using the geNomad_filter_outputs.sh code
```{bash script for summary directories, eval = FALSE}
#this shell script will take geNomad output files from genomad_output_final_indiv_assemblies (my genomad output directory holding all files) into their respective summary files
#in summary it will take separate summary files and combine them inot a big summary output file

#set up environment to run genomad
#source /programs/miniconda3/bin/activate genomad
#genomad download-database .

pwd #make sure that you are in the correct path /local/workdir/sna49/SA_Fall2023_Phage_Rotation/Data/final_indiv_assemblies/genomad_output_final_indiv_assemblies

#our results with the summary .tsv files are embedded in the output directory so we must get the .tsv files out of the directory. we do not want the log files.
#FIRST THINGS FIRST we want to get the files WITHIN the summary directories of each output.

#but we need to actually start by naming the actual directory where we want the results sent to
sum_directory="summary_tsv_files"

#now we will create the output directory
mkdir -p "sum_directory" #the -p ensures that there will be no error if there is an existing directory otherwise it will make the directory as needed

for d in *.fa_summary/ ; do #iterating thorugh directories ending in *.fa_summary/
  echo "$d" #printing each directory name with this ending
  find "$d" -type f -name "*.tsv" -exec cp {} "sum_directory/" \;
done

#now all files will be placed in /local/workdir/sna49/SA_Fall2023_Phage_Rotation/Data/final_indiv_assemblies/genomad_output_final_indiv_assemblies/sum_directory

#within that directory you will notice that all files have an ending with .tsv, but we will concatenate all files to end in their respective .txt file

cd indiv_sum_directory/ #make sure you are in sum_directory

#so now we will combine the output concatenated summary files into a long list and we will add a column that will specify the filename for each row

#so now we are using the awk command that processes certain patterns and actions
  #goes like: awk 'pattern { action }' input-file

#so you could make the .tsv files into a long list of .txt but I will be making them into .tsv files so I can analyze them in rstudio later 
#1. creating long list of all plasmid genes
awk 'NR==1{print $0" filename" }' *.fa_plasmid_genes.tsv > indiv_plasmid_genes.tsv
  #NR==1: checking to see if the current record number is 1, so it is processing first line aka the header of each file
  #print $0: filename: if the condition is met it will print the entire line ('$0') followed by space and string called "filename"
awk 'FNR>1{ sub(/\/[^\/]+$/, "", FILENAME); print FILENAME, $0 }' *.fa_plasmid_genes.tsv >> indiv_plasmid_genes.tsv
  #FNR>1: checks to see if current record number within the current file is greater than 1, skips the first line or the header of each input file
  #{ sub(/\/[^\/]+$/, "", FILENAME); print FILENAME, $0 }' : for reocrds that meet these conditions, it will modify the "filename" and remove teh directory path and then print modify 'filename' followed by entire line ('$0')
  
#2. creating long list of all plasmid summary
awk 'NR==1{print $0" filename" }' *.fa_plasmid_summary.tsv > indiv_plasmid_summary.tsv
awk 'FNR>1{ sub(/\/[^\/]+$/, "", FILENAME); print FILENAME, $0 }' *.fa_plasmid_summary.tsv >> indiv_plasmid_summary.tsv

#3. creating long list of all virus genes
awk 'NR==1{print $0" filename" }' *.fa_virus_genes.tsv > indiv_virus_genes.tsv
awk 'FNR>1{ sub(/\/[^\/]+$/, "", FILENAME); print FILENAME, $0 }' *.fa_virus_genes.tsv >> indiv_virus_genes.tsv
 
#4. creating long list of all virus summary
awk 'NR==1{print $0" filename" }' *.fa_virus_summary.tsv > indiv_virus_summary.tsv
awk 'FNR>1{ sub(/\/[^\/]+$/, "", FILENAME); print FILENAME, $0 }' *.fa_virus_summary.tsv >> indiv_virus_summary.tsv

#at this point we have successfully created our long lists of data of plasmid and phage information and we can now begin our Rstudio analysis

#then these 4 "concatenated"/ merged files are moved to indiv_sum_directory/4_cat_files

```

### Community analysis organizing concatenated outputs

```{r loading final individual summary files produced by geNomad}
#set working directory
setwd("/local/workdir/sna49/SA_Fall2023_Phage_Rotation/Data/final_indiv_assemblies/genomad_output_final_indiv_assemblies/indiv_sum_directory/4_cat_files")

#1. looking at plasmid genes
indiv_plasmid_genes <- read.table(file = "/local/workdir/sna49/SA_Fall2023_Phage_Rotation/Data/final_indiv_assemblies/genomad_output_final_indiv_assemblies/indiv_sum_directory/4_cat_files/indiv_plasmid_genes.tsv", sep ="\t", header = TRUE)

#2. looking at plasmid summary 
indiv_plasmid_summary <- read.table(file = "/local/workdir/sna49/SA_Fall2023_Phage_Rotation/Data/final_indiv_assemblies/genomad_output_final_indiv_assemblies/indiv_sum_directory/4_cat_files/indiv_plasmid_summary.tsv", sep ="\t", header = TRUE)

#3 looking at virus genes
indiv_virus_genes <- read.table(file = "/local/workdir/sna49/SA_Fall2023_Phage_Rotation/Data/final_indiv_assemblies/genomad_output_final_indiv_assemblies/indiv_sum_directory/4_cat_files/indiv_virus_genes.tsv",quote="", sep ="\t", header = TRUE)

#4 looking at virus summary
indiv_virus_summary <- read.table(file = "/local/workdir/sna49/SA_Fall2023_Phage_Rotation/Data/final_indiv_assemblies/genomad_output_final_indiv_assemblies/indiv_sum_directory/4_cat_files/indiv_virus_summary.tsv",quote="", sep ="\t", header = TRUE)

```

## Creating data frames so that we can analyze the geNomad outputs

### Creating dataframe of plasmid genes

```{r create dataframe of plasmid genes}
#1. plasmid genes
#need to create dataframe
df_indiv_plasmid_genes <- data.frame(indiv_plasmid_genes)

#split the first column called gene into multiple columns
#because there is so much going on with the column name we are going to separate each word into its own column and then mutate it together (dont be alarmed)
df_indiv_plasmid_genes <- df_indiv_plasmid_genes %>% separate(gene, c('Final', 'Seq_Name', 'Name','fa_info', 'plasmid', 'genes', 'tsv', 'Contig', 'bp_length', 'Contig_Number'))

#now we are going to mutate the combine the columns together



#then we need to get rid of the extra columns that we do not need
df_indiv_plasmid_genes <- subset(df_indiv_plasmid_genes, select = -c(Final,Name, fa_info, plasmid, genes, tsv, Contig, Contig_Number)) 
#view to make sure it looks good
```

### Creating dataframe of plasmid summary

```{r creating data frame of plasmid summary}
#2. plasmid summary
#need to create dataframe
df_indiv_plasmid_sum <- data.frame(indiv_plasmid_summary)

#split the first column called gene into multiple columns
df_indiv_plasmid_sum <- df_indiv_plasmid_sum %>% separate(seq_name, c('Final', 'Seq_Name', 'Name', 'fa_info', 'plasmid', 'genes', 'tsv', 'Contig', 'bp_length'))

#now we are going to mutate the combine the columns together
df_indiv_plasmid_sum$fa_info <- paste(df_indiv_plasmid_sum$fa_info, df_indiv_plasmid_sum$plasmid, df_indiv_plasmid_sum$genes, df_indiv_plasmid_sum$tsv, sep = "_")
df_indiv_plasmid_sum$Contig <- paste(df_indiv_plasmid_sum$Contig, df_indiv_plasmid_sum$bp_length, sep = "_")

#then we need to get rid of the extra columns that we do not need
df_indiv_plasmid_sum <- subset(df_indiv_plasmid_sum, select = -c(Final, Name, plasmid, genes, tsv)) 

#view to make sure it looks good

```

### Creating virus genes dataframe

```{r create virus genes dataframe}
#3. virus genes
#need to create dataframe
df_indiv_virus_genes <- data.frame(indiv_virus_genes)

#split the first column called gene into multiple columns
df_indiv_virus_genes <- df_indiv_virus_genes %>% separate(gene, c('Seq_Name', 'DASTool', 'Bin', 'fa_info', 'virus', 'genes', 'tsv', 'Contig', 'bp_length', 'Contig_Number'))

#now we are going to mutate the combine the columns together
df_indiv_virus_genes$Seq_Name <- paste(df_indiv_virus_genes$Seq_Name, df_indiv_virus_genes$DASTool, df_indiv_virus_genes$Bin, sep = "_")
df_indiv_virus_genes$fa_info <- paste(df_indiv_virus_genes$fa_info, df_indiv_virus_genes$virus, df_indiv_virus_genes$genes, df_indiv_virus_genes$tsv, sep = "_")
df_indiv_virus_genes$Contig <- paste(df_indiv_virus_genes$Contig, df_indiv_virus_genes$bp_length, sep = "_")

#then we need to get rid of the extra columns that we do not need
df_indiv_virus_genes <- subset(df_indiv_virus_genes, select = -c(DASTool, Bin, virus, genes, tsv))

#view to make sure it looks good
```

### Creating virus summary dataframe

```{r creating virus summary dataframe}
#4. virus summary
#need to create dataframe
df_indiv_virus_sum <- data.frame(indiv_virus_summary)

#split the first column called gene into multiple columns
df_indiv_virus_sum <- df_indiv_virus_sum %>% separate(seq_name, c('Seq_Name', 'Sample_ID', 'Bin', 'fa_info', 'virus', 'genes', 'tsv', 'Contig', 'bp_length'))

#now we are going to mutate the combine the columns together
df_indiv_virus_sum$fa_info <- paste(df_indiv_virus_sum$fa_info, df_indiv_virus_sum$virus, df_indiv_virus_sum$genes, df_indiv_virus_sum$tsv, sep = "_")
df_indiv_virus_sum$Contig <- paste(df_indiv_virus_sum$Contig, df_indiv_virus_sum$bp_length, sep = "_")

#then we need to get rid of the extra columns that we do not need
df_indiv_virus_sum <- subset(df_indiv_virus_sum, select = -c(Seq_Name, Bin, virus, genes, tsv, bp_length)) 

#view to make sure it looks good
```

#### title
```{r}
#now that we have finally parsed through our datafiles we want to create a figure that will show us how genome size is correlated to sample type. 
  # x axis: .22 um or 3.0
  # y axis: length/ Mbp

#according the meta file, the free samples are even and the particle samples are odd

#we will load the metafile into our working directory and then visualize the box plot 

#load in the metadata file 
assembly_meta <- read.csv("/local/workdir/sna49/SA_Fall2023_Phage_Rotation/Data/final_indiv_assemblies/metadata.csv")

#now lets subset the data
assembly_meta <- subset(assembly_meta, select = -c(names,lakesite,limnion,replicate,month,year,nuc_acid_type,project,season,names_nofracrep)) 

#now lets make a column that says if it is free = .22 μm or particle = 3.0 μm
assembly_meta <- assembly_meta %>% 
  mutate(filter_size = case_when(
    fraction == "Free" ~ "0.22 μm",
    fraction == "Particle" ~ "3.0 μm"
  ))

#ok now I am not sure what is the total length of the reads. Like is the sample_total_read_count just the number of reads? Or is it number of total reads * average length? 

#I think I will try multiplying the number of total seq reads by average length 
assembly_meta <- assembly_meta %>% 
  mutate(
    sample_total_read_count = as.numeric(sample_total_read_count),
    avg_read_length = as.numeric(avg_read_length),
    estimate_bp_length = sample_total_read_count * avg_read_length
    )

#then lets divide the estimate_bp_length by 1 million
assembly_meta <- assembly_meta %>% 
  mutate(length_mb = estimate_bp_length / 1000000)

#assuming everything is ok lets plot
mbp <- ggplot(assembly_meta, aes(x = filter_size, y = length_mb, fill = fraction))+
  geom_boxplot(outlier.shape = NA) +
  geom_jitter() +
  scale_fill_carto_d(palette = "Geyser")+
  theme_classic()

mbp

```

```{r normalizing plasmid and phage gene amounts that are found on each contig to a normalized bin abundance}
#individual asssembly normalizing

#before we normalize anything lets join our plasmid and virus sum dfs together with assembly meta

#first lets do plasmid

#need to make sure that assembly_meta is character
assembly_meta$sample_ID <- as.character(assembly_meta$sample_ID)

#now lets merge
df_meta_assembly_plasmid_sum <- left_join(df_indiv_plasmid_sum, assembly_meta, by = c("Seq_Name" = "sample_ID"))

#now lets group by sample_ID
df_meta_assembly_plasmid_sum <- df_meta_assembly_plasmid_sum %>% 
  group_by(Seq_Name, length_mb, fraction, filter_size) %>%
  summarise(count_n_genes = sum(n_genes))

#now we will divide the number of genes per Mb to get the abundance per assembly
df_meta_assembly_plasmid_sum$genes_per_Mb <- df_meta_assembly_plasmid_sum$count_n_genes / df_meta_assembly_plasmid_sum$length_mb 

#now lets visualize the abundance of plasmid encoded genes per contig 
assembly_plasmid_genes <- ggplot(df_meta_assembly_plasmid_sum, aes(x = filter_size, y = genes_per_Mb, fill = fraction)) +
  geom_boxplot() +
  geom_point() +
   geom_pwc(
    method = "wilcox_test", label = "p.adj.format",
    p.adjust.method = "fdr") +
  xlab("Filter Size") +
  ylab("Plasmid Encoded Genes / Mb") +
  theme_classic()
assembly_plasmid_genes
#p value is 0.51 for plasmid genes not significant but thats ok


#need to make sure that assembly_meta is character
assembly_meta$sample_ID <- as.character(assembly_meta$sample_ID)

#now lets merge
df_meta_assembly_virus_sum <- left_join(df_indiv_plasmid_sum, assembly_meta, by = c("Seq_Name" = "sample_ID"))
#now lets do the same for virus sum to look at number of phage encoded genes
df_meta_assembly_virus_sum <- left_join(df_indiv_virus_sum, assembly_meta, by = c("Sample_ID" = "sample_ID"))

#now lets group by sample_ID
df_meta_assembly_virus_sum <- df_meta_assembly_virus_sum %>% 
  group_by(Sample_ID, length_mb, fraction, filter_size) %>%
  summarise(count_n_genes = sum(n_genes))

#now we will divide the number of genes per Mb to get the abundance per assembly
df_meta_assembly_virus_sum$genes_per_Mb <- df_meta_assembly_virus_sum$count_n_genes / df_meta_assembly_virus_sum$length_mb 

#now lets visualize the abundance of plasmid encoded genes per contig 
assembly_phage_genes <- ggplot(df_meta_assembly_virus_sum, aes(x = fraction, y = genes_per_Mb, fill = fraction)) +
  geom_boxplot() +
  geom_point() +
   geom_pwc(
    method = "wilcox_test", label = "p.adj.format",
    p.adjust.method = "fdr") +
  xlab("Lifestyle") +
  ylab("# Phage-Encoded Genes / Mb")+
theme_classic()
assembly_phage_genes
#p value is 0.51 for plasmid genes not significant but thats ok
#group by sample id (with indiv_plasmid genes), ocunt n_genes and divide by assembly size to get number of genes per Mb and then plot
```

#### Conjugation Genes
```{r}
assembly_meta$sample_ID <- as.character(assembly_meta$sample_ID)

#now lets merge
df_meta_assembly_plasmid_genes <- left_join(df_indiv_plasmid_genes, assembly_meta, by = c("Seq_Name" = "sample_ID"))



#now we will make all conjugation type genes as an umbrella
community_conjugation <- df_meta_assembly_plasmid_genes %>% 
  filter(!is.na(annotation_conjscan)) %>%  
  mutate(conj_gene_type = case_when(grepl("trb", annotation_conjscan) ~ "trb genes",
                                    grepl("vir", annotation_conjscan) ~ "vir genes",
                                    grepl("MOB", annotation_conjscan) ~ "MOB genes",
                                    grepl("tra", annotation_conjscan) ~ "tra genes",
                                    grepl("t4cp1", annotation_conjscan) ~ "tra genes",
                                    grepl("alr", annotation_conjscan) ~ "trb genes")) %>% 
  select(Seq_Name, annotation_conjscan, conj_gene_type, filter_size, fraction, length_mb)

#now lets group by conjugation gene type
community_conjugation_type <- community_conjugation %>% 
  group_by(Seq_Name, length_mb, fraction, filter_size, conj_gene_type) %>%
  summarise(count_conjugation_genes = n())
conjugation_com <- df_indiv_plasmid_genes

community_conjugation_type$abundance <- community_conjugation_type$count_conjugation_genes / community_conjugation_type$length_mb

all_conj_genes <- community_conjugation %>% 
  group_by(Seq_Name, filter_size, length_mb) %>%
  summarise(count_allconjugation_genes = n())

all_conj_genes$abundance <- all_conj_genes$count_allconjugation_genes / all_conj_genes$length_mb

#plot all conjugation genes
all_conjugation <- ggplot(all_conj_genes, aes(x = filter_size, y = abundance, fill = filter_size)) +
  geom_boxplot() +
  geom_point() +
   geom_pwc(
    method = "wilcox_test", label = "p.adj.format",
    p.adjust.method = "fdr") +
  xlab("Filter Size") +
  ylab("Conjugation Genes / Mb") +
  theme_classic()
all_conjugation

#now lets look at specific conjugation genes
#plot all conjugation genes
conjugation_type <- ggplot(community_conjugation_type, aes(x = filter_size, y = abundance, fill = conj_gene_type)) +
  geom_bar(stat = "identity", position = "dodge") +
  scale_fill_wa_d(palette = "rainier", reverse = TRUE)+
  xlab("Filter Size") +
  ylab("Conjugation Genes / Mb") +
  theme_classic()
conjugation_type
  
```

